### PR TITLE
Wasm switch

### PR DESCRIPTION
**Pull request by chriseth**
Depends on #2405

### CODE REVIEW

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `898ea3ba602f6a30a371890db81045750d856e18`:
```
Coding style error:
 libyul/backends/wasm/WordSizeTransform.cpp:232: (valueOfLiteral(*c.value) >> (256 - 64 CMakeLists.txt CODE_OF_CONDUCT.md CODING_STYLE.md CONTRIBUTING.md Changelog.md LICENSE.txt README.md ReleaseChecklist.md appveyor.yml cmake codecov.yml docs libdevcore libevmasm liblangutil liblll libsolc libsolidity libyul lllc scripts snap solc test (_depth + 1))) &

```
Please check that your changes are working as intended.

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `e7f2586def686b27254ee0d23cb582318f08b570`:
```
Coding style error:
 libyul/backends/wasm/WordSizeTransform.cpp:232: (valueOfLiteral(*c.value) >> (256 - 64 CMakeLists.txt CODE_OF_CONDUCT.md CODING_STYLE.md CONTRIBUTING.md Changelog.md LICENSE.txt README.md ReleaseChecklist.md appveyor.yml cmake codecov.yml docs libdevcore libevmasm liblangutil liblll libsolc libsolidity libyul lllc scripts snap solc test (_depth + 1))) &

```
Please check that your changes are working as intended.

File: test/libyul/yulOptimizerTests/wordSizeTransform/switch_1.yul [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+}
+// ====
+// step: wordSizeTransform
+// ----
+// {
+//     let _1_0 := 0
+//     let _1_1 := 0
+//     let _1_2 := 0
+//     let _1_3 := 0
+//     let _2_0, _2_1, _2_2, _2_3 := calldataload(_1_0, _1_1, _1_2, _1_3)
```
Comment (leonardoalt) [inline test/libyul/yulOptimizerTests/wordSizeTransform/switch_1.yul:16]: Does `calldataload/4` exist?

Comment (chriseth) [reply test/libyul/yulOptimizerTests/wordSizeTransform/switch_1.yul:16]: In this dialect, yes

File: libyul/backends/wasm/WordSizeTransform.h [hunk 2]
```diff
@@ -78,6 +80,15 @@ class WordSizeTransform: public ASTModifier
 	void rewriteIdentifierList(std::vector<Identifier>&);
 	void rewriteFunctionCallArguments(std::vector<Expression>&);
 
+	std::vector<Statement> handleSwitch(Switch& _switch);
+	std::vector<Statement> handleSwitchInternal(
+		langutil::SourceLocation _location,
```
Comment (leonardoalt) [inline libyul/backends/wasm/WordSizeTransform.h]:
```suggestion
		langutil::SourceLocation const& _location,
```
?

**Comment by chriseth** [comment]
Rebased.

**Comment by chriseth** [comment]
Updated.

**Review by leonardoalt — APPROVED** [review]
The segment grouping looks nice

**Pull request merged**